src/drive/targets/public/index.jsx

```diff
@@ -115,17 +117,23 @@ const init = async () => {
         <App lang={lang} polyglot={polyglot} client={client} store={store}>
           {isFile ? (
             <Router history={hashHistory}>
-              <Route
-                path="/"
-                component={() => (
-                  <PublicLayout>
-                    <LightFileViewer files={[data]} />
-                  </PublicLayout>
+              <Route component={PublicLayout}>
+                {isOnlyOfficeEnabled() && (
+                  <Route
+                    path="onlyoffice/:fileId"
+                    component={props => (
+                      <OnlyOfficeView {...props} isPublic={true} />
+                    )}
+                  />
+                )}
+                {shouldBeOpenedByOnlyOffice(data) && (
+                  <Redirect from="/" to={`onlyoffice/${data.id}`} />
```

Comment:
With this redirection, we go to /onlyoffice/..., but this it's a redirection, the check to know if we should display the back button will return true, no?

Reply:
Non c'est ok, <Redirect /> n'ajoute pas d'entrée dans l'history mais remplace l'existante https://reactrouter.com/web/api/Redirect (fonctionnement identique sur notre version v3)